### PR TITLE
Redefine minVisiblePercentage and maxVisiblePercentage behavior after visibility reset

### LINKED ISSUE #12529: Discuss: meaning of min/maxVisibleTime variable substitution
If an amp-analytics trigger uses ${minVisiblePercentage} and/or ${maxVisiblePercentage} substitution variables, currently the values that are substituted for these are a bit unintuitive.

Example:
Have a trigger with visiblePercentageThresholds: [[0,100]], repeat: true.
Hit reload when ad is 18% visible. Trigger fires w/ maxVisiblePercentage = 18%.
Scroll to 40% visible. (No event fires, since we didn't fall out of visibility.)
Scroll to 0% visible. (No event fires. reset() is called on the VisibilityModel.)
Scroll to 5% visible. Trigger fires w/ maxVisiblePercentage = 5%.
Scroll to more visible.  (No event fires, since we didn't fall out of visibility.)
The above steps are shown at: https://youtu.be/WQ_l1jsMeRA

What maxVisiblePercentage is really showing is "the maximum visible percentage since the last time reset() was called" which, without continuousTimeMin being set, is effectively just "whatever % visible the ad first was, the most recent time it was scrolled from 0% visible to a non-zero visibility." In this example, "visibilityRangeEntryPoint" would be a more apt name, though probably not very useful.

The current name+behaviour is unintuitive, since you don't expect something called "maximum" to go up and down.

Example 2:
Same as before, but add "continuousTimeMin": 5000
Hit reload when ad is 44% visible. Trigger fires w/ maxVisiblePercentage = 44%.
Scroll to 60% visible. Wait 5 seconds for reset() to be called. Trigger fires w/ maxVisiblePercentage = 60%.
Scroll to 75% visible. Wait 5 seconds for reset() to be called. Trigger fires w/ maxVisiblePercentage = 75%.
Scroll to 20% visible. Wait 5 seconds for reset() to be called. Trigger fires w/ maxVisiblePercentage = 20%.
The above steps are shown at: https://youtu.be/lJmYeXiPbK0

Now maxVisiblePercentage = "the visible percentage the last time the continuousTimeMin interval reset (...or the ad was scrolled out of view and back into view)".

As before, this is unintuitive since the reported maximum can decrease.

I wonder if we should rename the min/maxVisibleTime variables to more accurately reflect what they do, or if we should stop setting them to 0 in VisibilityModel.reset().

### PR DESCRIPTION
**Pull request by zhouyx**
Closes #12529

### CODE REVIEW

File: extensions/amp-analytics/amp-analytics.md [hunk 1]
```diff
@@ -578,7 +578,7 @@ The `visibilitySpec` is a set of conditions and properties that can be applied t
...
-  - `repeat`: If this property is set to `true`, the trigger fires each time that the `visibilitySpec` conditions are met. In the following example, if the element is scrolled to 51% in view, then 49%, then 51% again, the trigger fires twice. However, if `repeat` was `false`, the trigger fires once. The default value of `repeat` is `false`. `reportWhen` and `repeat` may not both be used in the same visibilitySpec.
+  - `repeat`: If this property is set to `true`, the trigger fires each time that the `visibilitySpec` conditions are met. In the following example, if the element is scrolled to 51% in view, then 49%, then 51% again, the trigger fires twice. However, if `repeat` was `false`, the trigger fires once. The default value of `repeat` is `false`. `reportWhen` and `repeat` may not both be used in the same visibilitySpec. Once the condition has been met, a list of visibility event [variables](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md#visibility) will be reset. This includes `${firstSeenTime}`, `${lastSeenTime}`, `${firstVisibleTime}`, `${lastVisibleTime}`, `${maxContinuousVisibleTime}` and `${totalVisibleTime}`.
```
Comment (lannka) [inline extensions/amp-analytics/amp-analytics.md]: what do you think about those metrics being reset? how different are they from `maxVisiblePercentage` & `minVisiblePercentage`?

Comment (zhouyx) [reply extensions/amp-analytics/amp-analytics.md]: good point. We could choose not to reset any metric above. However by doing that publishers are losing a lot of metrics that were previously provided. 
We should maybe consider `${xxx}` and `${localXXX}` to clarify the difference between global and local metrics. 
I'll put this PR on hold, until I get back from @jonkeller. Jon any feedback on what the metrics should be when we send multiple visibility requests.  Thank you

Comment (lannka) [reply extensions/amp-analytics/amp-analytics.md]: true. let's be a bit cautious.

Comment (jonkeller) [reply extensions/amp-analytics/amp-analytics.md]: I would prefer to not reset these variables. The localXXX idea sgtm, though. Thanks!

**Comment by lannka** [comment]
closing this for inactivity. feel free to reopen

**Pull request closed without merging**